### ISSUE TITLE
Decision task ignores TASKCLUSTER_HEAD_REV

### ISSUE THREAD

**Issue by petemoore** [taskcluster/taskcluster#3145]
**Describe the bug**
Sometimes links to taskcluster CI tasks from github PR UI are for the wrong commit.

**To Reproduce**

1. Go to https://github.com/taskcluster/taskcluster/pull/3103
2. Click on the green tick next to commit bb91dd2
3. Click on "Details" next to task "Build/test generic-worker (multiuser engine) on Windows Server 2012 R2 (amd64)" (note this is the third taskcluster task listed when I view it, ymmv)
4. You should be taken to https://github.com/taskcluster/taskcluster/runs/830725558
5. Note that this page of the UI correctly refers to commit bb91dd2 at the top of the page. So far so good.
6. Note that the correct task is selected with a blue background in the left column on the page ("Build/test generic-worker (mult...")
7. Click on "View task in Taskcluster"
8. You should be taken to [task F8vva6ToTuCAourtdTiktw](https://community-tc.services.mozilla.com/tasks/F8vva6ToTuCAourtdTiktw).  
9. Note the task has the correct title "Build/test generic-worker (multiuser engine) on Windows Server 2012 R2 (amd64)".
10. Click on "Payload"
11. Note that the payload contains `"GITHUB_SHA": "41acdacab93f73704f86069727664b61b0144072"`. This is a different commit! Looking at it, we see it is a _later_ commit, since the commit we expected to see (bb91dd2) is actually its parent:

```
$ git log -n1 --pretty=raw 41acdacab93f73704f86069727664b61b0144072
commit 41acdacab93f73704f86069727664b61b0144072
tree 35fcd0773b4b6ec830d94f27a359698f17cc849c
parent bb91dd2920ed3e94ea2256c2911f68baf306cb74
author Pete Moore <pmoore@mozilla.com> 1593626808 +0200
committer Pete Moore <pmoore@mozilla.com> 1593695077 +0200
gpgsig -----BEGIN PGP SIGNATURE-----
 Comment: GPGTools - http://gpgtools.org
 
 iQFHBAABCAAxFiEEhe6Nva+1y60nOjNRgpX+d7R1/rwFAl7922UTHHBtb29yZUBt
 b3ppbGxhLmNvbQAKCRCClf53tHX+vERFB/9SeEGHJpBdmCTiYkLbxFOYth8c8JAw
 uskZ9r5WPBHmVKbX0BvJTbaya8BBidbQ5i3j/QN+JlZCWAFI7i7PL2uGrHWvdDr+
 bUkDxEZ8MjblE7LpbTn/2iMdrG7SggJ00JbS2dy8whtQuBm4gO2deuGtYLdjTaT2
 Pk+2jjzzcXR5DXkc5tWKgvz9N6zQjP15Xpqqlst0b61z4DoXD3rrT11qnjc25KDj
 B+eb69ljVSiyL3fQz76zhveuZiq/cneGwtK6B1dR7s/CAx1y5kcMs0cpAVe4PzK7
 AUN8rYhL7hF+xUtRPIOw2ZedmEu/ZcbRVDYkcn6517mR0fO9FJjCbNR5
 =8q+w
 -----END PGP SIGNATURE-----

    Issue 2934: Part 3 - Skip mock db tests (issue #3131)
```

**Expected behavior**
The link tasks linked against a given commit in a github PR should be the task created for that commit for that PR.

**Taskcluster version**
31.0.0

**Comment by djmitche** [comment]
The [decision task](https://community-tc.services.mozilla.com/tasks/DqEkSiCISueSG6CY55_dvg) in [that task group](https://community-tc.services.mozilla.com/tasks/groups/DqEkSiCISueSG6CY55_dvg) ran for the correct revision:
```
    "TASKCLUSTER_HEAD_REV": "bb91dd2920ed3e94ea2256c2911f68baf306cb74",
```

but [`public/parameters.yml`](https://community-tc.services.mozilla.com/api/queue/v1/task/DqEkSiCISueSG6CY55_dvg/runs/0/artifacts/public/parameters.yml) the incorrect rev.  I suspect there's something misconfigured with taskgraph here.

**Comment by djmitche** [comment]
Even the clone for the decision task is wrong, cloning the latest version of the ref instead of the given rev:
```
[vcs 2020-07-02T13:11:04.949Z] executing ['git', 'clone', 'https://github.com/taskcluster/taskcluster', '/builds/worker/checkouts/src']
[vcs 2020-07-02T13:11:04.952Z] Cloning into '/builds/worker/checkouts/src'...
[vcs 2020-07-02T13:11:15.149Z] executing ['git', 'fetch', 'https://github.com/taskcluster/taskcluster', 'issue2934']
[vcs 2020-07-02T13:11:15.374Z] From https://github.com/taskcluster/taskcluster
[vcs 2020-07-02T13:11:15.374Z]  * branch                issue2934  -> FETCH_HEAD
[vcs 2020-07-02T13:11:15.385Z] executing ['git', 'checkout', '-f', 'FETCH_HEAD']
[vcs 2020-07-02T13:11:15.653Z] Note: checking out 'FETCH_HEAD'.
[vcs 2020-07-02T13:11:15.653Z] 
[vcs 2020-07-02T13:11:15.653Z] You are in 'detached HEAD' state. You can look around, make experimental
[vcs 2020-07-02T13:11:15.653Z] changes and commit them, and you can discard any commits you make in this
[vcs 2020-07-02T13:11:15.653Z] state without impacting any branches by performing another checkout.
[vcs 2020-07-02T13:11:15.653Z] 
[vcs 2020-07-02T13:11:15.653Z] If you want to create a new branch to retain commits you create, you may
[vcs 2020-07-02T13:11:15.653Z] do so (now or later) by using -b with the checkout command again. Example:
[vcs 2020-07-02T13:11:15.653Z] 
[vcs 2020-07-02T13:11:15.653Z]   git checkout -b <new-branch-name>
[vcs 2020-07-02T13:11:15.653Z] 
[vcs 2020-07-02T13:11:15.653Z] HEAD is now at 41acdacab Issue 2934: Part 3 - Skip mock db tests (issue #3131)
```

Looking at [the code](https://hg.mozilla.org/ci/taskgraph/file/tip/src/taskgraph/run-task/run-task#l676) I would expect `options['revision']` to be `bb91dd2920ed3e94ea2256c2911f68baf306cb74` and `options['ref']` to be `issue2932`.  Those are [passed through to git_checkout](https://hg.mozilla.org/ci/taskgraph/file/tip/src/taskgraph/run-task/run-task#l744) which runs [`git fetch <repo> <ref>`](https://hg.mozilla.org/ci/taskgraph/file/tip/src/taskgraph/run-task/run-task#l442) and we see that in the log output above.  It next runs [`git checkout -f <rev>`](https://hg.mozilla.org/ci/taskgraph/file/tip/src/taskgraph/run-task/run-task#l451), but that seems to translate to `['git', 'checkout', '-f', 'FETCH_HEAD']` above.  That suggests that the expression `commit if commit else ref` evaluated to `FETCH_HEAD`.  Since rev was used in the previous command and has value `issue2932`, that means that `commit` must have the value `FETCH_HEAD`.  But I don't see where that string can have come from - it's not in the task payload, and it's not anywhere in the taskgraph code or in `taskcluster/` in the taskcluster monorepo.

@tomprince I'm obviously missing something -- any chance it's obvious to you?

**Comment by tomprince** [comment]
It looks like `run-task` was fixed in [Bug 1595808](https://bugzilla.mozilla.org/show_bug.cgi?id=1595808) but the decision image on docker hub was not updated with that fix.
